app/logic/courseManagement.py

```diff
@@ -44,41 +45,52 @@ def approvedCourses(termId):
-        term=courseData['term'],
-        status=CourseStatus.SUBMITTED,
-        isAllSectionsServiceLearning=("on" in courseData["slSectionsToggle"]),
-        serviceLearningDesignatedSections=courseData["slDesignation"],
-        isPermanentlyDesignated=("on" in courseData["permanentDesignation"]),
-    ).where(Course.id == course.id).execute()
-    for i in range(1, 7):
-        (CourseQuestion.update(questionContent=courseData[f"{i}"])
-                    .where((CourseQuestion.questionNumber == i) &
-                           (CourseQuestion.course==course)).execute())
-    instructorList = []
-    if 'instructor[]' in courseData:
-        instructorList = courseData.getlist('instructor[]')
-    CourseInstructor.delete().where(CourseInstructor.course == course).execute()
-    for instructor in instructorList:
-        CourseInstructor.create(course=course, user=instructor)
-    createLog(f"Saved SLC proposal: {courseData['courseName']}")
-    return Course.get_by_id(course.id)
+    try:
+        course = Course.get_by_id(courseData['courseID'])
+        for toggler in ["slSectionsToggle", "permanentDesignation"]:
+            courseData.setdefault(toggler, "off")
+        Course.update(
+            courseName=courseData["courseName"],
+            courseAbbreviation=courseData["courseAbbreviation"],
+            sectionDesignation=courseData["sectionDesignation"],
+            courseCredit=courseData["credit"],
+            isRegularlyOccurring=int(courseData["isRegularlyOccurring"]),
+            term=courseData['term'],
+            status=CourseStatus.SUBMITTED,
+            isPreviouslyApproved=int(courseData["isPreviouslyApproved"]),
+            previouslyApprovedDescription = courseData["previouslyApprovedDescription"],
+            isAllSectionsServiceLearning=("on" in courseData["slSectionsToggle"]),
+            serviceLearningDesignatedSections=courseData["slDesignation"],
+            isPermanentlyDesignated=("on" in courseData["permanentDesignation"]),
+            hasSlcComponent = int(courseData["hasSlcComponent"])
+        ).where(Course.id == course.id).execute()
+        for i in range(1, 7):
+            (CourseQuestion.update(questionContent=courseData[f"{i}"])
+                        .where((CourseQuestion.questionNumber == i) &
+                            (CourseQuestion.course==course)).execute())
+        instructorList = []
+        if 'instructor[]' in courseData:
+            instructorList = courseData.getlist('instructor[]')
+        CourseInstructor.delete().where(CourseInstructor.course == course).execute()
+        for instructor in instructorList:
+            CourseInstructor.create(course=course, user=instructor)
+        createLog(f"Saved SLC proposal: {courseData['courseName']}")
+        if attachment:
+            addfile= FileHandler(attachment, courseId=course.id)
+            addfile.saveFiles()
+        return Course.get_by_id(course.id)
+    except Exception as e:
```

Comment:
Given that we are modifying course, instructors, questions, and files, it would be a good idea to add a transaction around this, and roll back in the `except`